migrator/src/main/java/com/google/phonenumbers/migrator/CommandLineMain.java

```diff
@@ -0,0 +1,102 @@
+package com.google.phonenumbers.migrator;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Scanner;
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Help.Ansi;
+import picocli.CommandLine.Option;
+
+@Command(name = "Migrator Tool", description = "Command line phone number migration tool.\n"
+    + "Please enter a path to a text file containing E.164 phone numbers from the same region or a "
+    + "single E.164 number as well as the corresponding region code to begin migrations.\n")
+public class CommandLineMain {
+
+  @Option(names = {"-n", "--number"}, description = "Single E.164 phone number to migrate")
+  String numberInput;
+
+  @Option(names = {"-f",
+      "--file"}, description = "Text file containing E.164 phone numbers to migrate")
+  String fileInput;
+
+  @Option(names = {"-r",
+      "--region"}, description = "The two digit BCP-47 region code the given phone number(s) belong to")
+  String regionCode;
+
+  @Option(names = {"-h", "--help"}, description = "Display help", usageHelp = true)
+  boolean help;
+
+  public String getNumberInput() {
+    return numberInput;
+  }
+
+  public String getFileInput() {
+    return fileInput;
+  }
+
+  public String getRegionCode() {
+    return regionCode;
+  }
+
+  public boolean insufficientArguments() {
+    return regionCode == null || (numberInput == null && fileInput == null);
+  }
+
+
+  /**
+   * Runs the command line migrator tool with functionality specified by then given user's command
+   * line arguments
+   *
+   * @param args which expects two command line arguments; numberInput: single E.164 number string
```

Comment:
nit: putting this on a new line would make the arguments easier to read.

Reply:
For this I'd document the args in the main block above.

Maybe using an unordered list (HTML):
----
<ul>
  <li>numberInput: ...
  <li>regionCode: ...
</ul>
----
Or the standard unix pseudo syntax for command lines, something like:

<command> ( numberInput | fileInput ) regionCode

The @param is a JavaDoc thing for callers from Java code, and a main() method is called by the JVM, so I wouldn't even have a @param for main args at all.



The "regionCode:" bit is odd though, and once you have a @param you should not have new paragraphs below it.

Reply:
I've just realised the Intellij formatting has messed up what the comment originally looked like and has merged it all into a single paragraph. I will re-format it and then also add it to the the main code block above.